pkg/component/runtime/command_test.go

```diff
@@ -0,0 +1,46 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package runtime
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddToBucket(t *testing.T) {
+	testCases := map[string]struct {
+		bucketSize  int
+		add         int
+		addSleep    time.Duration
+		shouldBlock bool
+	}{
+		"no error":           {1, 0, 10 * time.Millisecond, false},
+		"error within limit": {1, 1, 10 * time.Millisecond, false},
+		"errors > than limit but across timespans":                {1, 2, 800 * time.Millisecond, false},
+		"errors > than limit within timespans, exact bucket size": {2, 2, 20 * time.Millisecond, false},
+		"errors > than limit within timespans, off by one":        {2, 3, 20 * time.Millisecond, true},
+		"errors > than limit within timespans":                    {2, 4, 20 * time.Millisecond, true},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			dropRate := 500 * time.Millisecond
```

Comment:
Consider using shorter intervals, this entire test suite takes 6s to run which seems excessive:

```
ok      github.com/elastic/elastic-agent/pkg/component/runtime  6.292s
```

Dividing everything by 10 works (so does 100 but maybe more chance of test flakiness in CI with some 100 microsecond intervals):

```diff
diff --git a/pkg/component/runtime/command_test.go b/pkg/component/runtime/command_test.go
index ef361b33b..ca4ee184c 100644
--- a/pkg/component/runtime/command_test.go
+++ b/pkg/component/runtime/command_test.go
@@ -18,22 +18,22 @@ func TestAddToBucket(t *testing.T) {
                addSleep    time.Duration
                shouldBlock bool
        }{
-               "no error":           {1, 0, 10 * time.Millisecond, false},
-               "error within limit": {1, 1, 10 * time.Millisecond, false},
-               "errors > than limit but across timespans":                {1, 2, 800 * time.Millisecond, false},
-               "errors > than limit within timespans, exact bucket size": {2, 2, 20 * time.Millisecond, false},
-               "errors > than limit within timespans, off by one":        {2, 3, 20 * time.Millisecond, true},
-               "errors > than limit within timespans":                    {2, 4, 20 * time.Millisecond, true},
+               "no error":           {1, 0, 1 * time.Millisecond, false},
+               "error within limit": {1, 1, 1 * time.Millisecond, false},
+               "errors > than limit but across timespans":                {1, 2, 80 * time.Millisecond, false},
+               "errors > than limit within timespans, exact bucket size": {2, 2, 2 * time.Millisecond, false},
+               "errors > than limit within timespans, off by one":        {2, 3, 2 * time.Millisecond, true},
+               "errors > than limit within timespans":                    {2, 4, 2 * time.Millisecond, true},
        }

        for name, tc := range testCases {
                t.Run(name, func(t *testing.T) {
-                       dropRate := 500 * time.Millisecond
+                       dropRate := 50 * time.Millisecond
                        b := newRateLimiter(dropRate, tc.bucketSize)

                        blocked := false
                        b.Allow()
-                       <-time.After(dropRate + 200*time.Millisecond) // init ticker
+                       <-time.After(dropRate + 20*time.Millisecond) // init ticker

                        for i := 0; i < tc.add; i++ {
                                wasBlocked := !b.Allow()
```

This gives `ok      github.com/elastic/elastic-agent/pkg/component/runtime  0.909s
`